pkg/appmgmt/general/podevent_handler.go

```diff
@@ -127,6 +128,13 @@ func (p *PodEventHandler) addPod(pod *v1.Pod, eventSource EventSource) interface
 				Metadata: appMeta,
 			})
 		} else {
+			if conf.GetSchedulerConf().GetAllowSimilarAppIdsByDifferentUsers() && app.GetApplicationState() == "Running" && app.GetUser() != appMeta.User {
+				log.Log(log.ShimAppMgmtGeneral).Warn("application has been submitted by different user",
+					zap.String("app id ", appMeta.ApplicationID),
+					zap.String("app user", app.GetUser()),
+					zap.String("submitted by", appMeta.User))
+				return nil
```

Comment:
What happens if we just return here? Does the pod remain in Pending state?

Reply:
We should write out an event on the pod using the `events.GetRecorder().Eventf(...)`
We do that in other places in our code that makes it traceable from the K8s API on the pod.

Reply:
+1 we have to do something so that users can see that something happened and they can act accordingly